### ISSUE TITLE
some validators are penalized while some are rewarded

### ISSUE THREAD

**Issue by somuchid** [prysmaticlabs/prysm#4589]
this is what happen on my  9 validators:
1. sometime are all good, rewarded
2. sometimes are all bad, penalized
3. sometimes a few are rewarded, the others are penalized
4. some validators receive assignment but have not submitted attestations

below are 4 cases just happened, try to find linkage between validator process and beacon node process:

-----------------------------------------------------------------------------------------
case 1: no error, all submit fine
-----------------------------------------------------------------------------------------
```
[2020-01-19 07:40:04]  INFO blockchain: Finished applying state transition attestations=6 deposits=0 slot=68600
[2020-01-19 07:40:15]  INFO forkchoice: Executing state transition on block root=0x7beb872c... slot=68601
[2020-01-19 07:40:16]  INFO blockchain: Finished applying state transition attestations=6 deposits=0 slot=68601
[2020-01-19 07:40:27]  INFO forkchoice: Executing state transition on block root=0xbb3dc445... slot=68602

[2020-01-19 07:35:24]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[16802] BeaconBlockRoot=0x9b2bf1ce9d56 CommitteeIndex=4 Slot=68576 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x9b2bf1ce9d56
[2020-01-19 07:36:24]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[17600] BeaconBlockRoot=0x0b947a704f25 CommitteeIndex=2 Slot=68581 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x43a7b79a0b05
[2020-01-19 07:37:00]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[18030] BeaconBlockRoot=0x939446baacfa CommitteeIndex=2 Slot=68584 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x43a7b79a0b05
[2020-01-19 07:37:00]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[18153] BeaconBlockRoot=0x939446baacfa CommitteeIndex=5 Slot=68584 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x43a7b79a0b05
[2020-01-19 07:37:48]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[18093] BeaconBlockRoot=0x735866e41b3c CommitteeIndex=2 Slot=68588 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x43a7b79a0b05
[2020-01-19 07:38:24]  INFO validator: Submitted new attestations AggregatorIndices=[16693] AttesterIndices=[16693] BeaconBlockRoot=0xf8c2dde2a720 CommitteeIndex=4 Slot=68591 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x43a7b79a0b05
[2020-01-19 07:38:36]  INFO validator: Submitted new attestations AggregatorIndices=[16525] AttesterIndices=[16525] BeaconBlockRoot=0x80ef8fb50150 CommitteeIndex=4 Slot=68592 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x43a7b79a0b05
[2020-01-19 07:39:00]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[17918] BeaconBlockRoot=0x9b3035b317ca CommitteeIndex=4 Slot=68594 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x43a7b79a0b05
[2020-01-19 07:40:12]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[16906] BeaconBlockRoot=0x11d2f13cf4fd CommitteeIndex=2 Slot=68600 SourceEpoch=2142 SourceRoot=0x4b559ff48792 TargetEpoch=2143 TargetRoot=0x43a7b79a0b05

[2020-01-19 07:41:36]  INFO validator: New Balance epoch=2143 newBalance=3.139709454 percentChange=-0.00003% prevBalance=3.139710435 pubKey=0x8a3888ca69e0d039
```
-----------------------------------------------------------------------------------------
case 2: error, 2 didn't submit attestation
-----------------------------------------------------------------------------------------
```
[2020-01-19 07:41:36]  INFO validator: New assignment attesterSlot=68617 committeeIndex=3 epoch=2144 pubKey=0x876920f4a991 status=ACTIVE validatorIndex=16802

-- 9 validators, only 7 submit
[2020-01-19 07:42:00]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[17600] BeaconBlockRoot=0xc62f14b20b94 CommitteeIndex=2 Slot=68609 SourceEpoch=2143 SourceRoot=0x43a7b79a0b05 TargetEpoch=2144 TargetRoot=0xc62f14b20b94
[2020-01-19 07:43:00]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[16525] BeaconBlockRoot=0x9dfd9eba4845 CommitteeIndex=0 Slot=68614 SourceEpoch=2143 SourceRoot=0x43a7b79a0b05 TargetEpoch=2144 TargetRoot=0xc62f14b20b94
[2020-01-19 07:43:12]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[16693] BeaconBlockRoot=0x2790c16addbc CommitteeIndex=0 Slot=68615 SourceEpoch=2143 SourceRoot=0x43a7b79a0b05 TargetEpoch=2144 TargetRoot=0xc62f14b20b94
[2020-01-19 07:43:36]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[16802] BeaconBlockRoot=0xdb796dd17b7b CommitteeIndex=3 Slot=68617 SourceEpoch=2143 SourceRoot=0x43a7b79a0b05 TargetEpoch=2144 TargetRoot=0xc62f14b20b94
[2020-01-19 07:43:36]  INFO validator: Submitted new attestations AggregatorIndices=[16906] AttesterIndices=[16906] BeaconBlockRoot=0xdb796dd17b7b CommitteeIndex=1 Slot=68617 SourceEpoch=2143 SourceRoot=0x43a7b79a0b05 TargetEpoch=2144 TargetRoot=0xc62f14b20b94
[2020-01-19 07:44:24]  INFO validator: Submitted new attestations AggregatorIndices=[17918] AttesterIndices=[17918] BeaconBlockRoot=0xc28417a5c8f9 CommitteeIndex=0 Slot=68621 SourceEpoch=2143 SourceRoot=0x43a7b79a0b05 TargetEpoch=2144 TargetRoot=0xc62f14b20b94
[2020-01-19 07:47:48]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[18153] BeaconBlockRoot=0x676efedeeb40 CommitteeIndex=5 Slot=68638 SourceEpoch=2143 SourceRoot=0x43a7b79a0b05 TargetEpoch=2144 TargetRoot=0xc62f14b20b94

[2020-01-19 07:47:36] ERROR blockchain: Could not receive attestation in chain service error=signature did not verify

[2020-01-19 07:48:00]  INFO validator: New Balance epoch=2144 newBalance=3.140166888 percentChange=-0.00003% prevBalance=3.14016787 pubKey=0x876920f4a9914d24
```
-----------------------------------------------------------------------------------------
case 3: error
-----------------------------------------------------------------------------------------
```
[2020-01-19 07:48:00]  INFO validator: New assignment attesterSlot=68666 committeeIndex=4 epoch=2145 pubKey=0xb97be494fb17 status=ACTIVE validatorIndex=17918

[2020-01-19 07:52:36] ERROR blockchain: Could not receive attestation in chain service error=signature did not verify

[2020-01-19 07:52:36]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[16525] BeaconBlockRoot=0xc1a043a35352 CommitteeIndex=1 Slot=68662 SourceEpoch=2144 SourceRoot=0xc62f14b20b94 TargetEpoch=2145 TargetRoot=0x2f7e0ee25bc8

[2020-01-19 07:54:24]  INFO validator: New Balance epoch=2145 newBalance=3.145477056 percentChange=-0.00054% prevBalance=3.145494006 pubKey=0xb97be494fb170972
```
-----------------------------------------------------------------------------------------
case 4: error
-----------------------------------------------------------------------------------------
```
[2020-01-19 07:54:24]  INFO validator: New assignment attesterSlot=68701 committeeIndex=0 epoch=2146 pubKey=0xb050c50a366e status=ACTIVE validatorIndex=18030

[2020-01-19 07:59:36]  INFO validator: Submitted new attestations AggregatorIndices=[] AttesterIndices=[18153] BeaconBlockRoot=0xb6329cf5f030 CommitteeIndex=0 Slot=68697 SourceEpoch=2145 SourceRoot=0x2f7e0ee25bc8 TargetEpoch=2146 TargetRoot=0xdf64d554ce46

[2020-01-19 07:59:36] ERROR blockchain: Could not receive attestation in chain service error=signature did not verify

[2020-01-19 08:00:48]  INFO validator: New Balance epoch=2146 newBalance=3.147274905 percentChange=-0.00003% prevBalance=3.147275917 pubKey=0xb050c50a366ed989
```

**Comment by somuchid** [comment]
as you can see, it is hard to link validator's output with beacon node's, and it is also hard to follow the info follow in validator, due to lack of a reference.

Learning from relational database Primary Key, I suggest adding public key as reference, in beacon node's error message relate to validator, also in "submit attestion" message of validator. Public key is already shown in validator's "new assignment" and "new balance".

**Comment by Buttaa** [comment]
this sounds like this (?) https://github.com/prysmaticlabs/prysm/issues/4569#issue-551112826

**Comment by Buttaa** [comment]
![grafik](https://user-images.githubusercontent.com/26490734/72897771-05485e80-3d23-11ea-8ec8-81cbeff5c5d0.png)
![grafik](https://user-images.githubusercontent.com/26490734/72897775-0a0d1280-3d23-11ea-927b-3e99f2d74683.png)
![grafik](https://user-images.githubusercontent.com/26490734/72897779-0c6f6c80-3d23-11ea-8d26-e922f253215c.png)

There has to be a bug with attestations since I was punished even though I was online. 
I hope the screenies help.

**Comment by somuchid** [comment]
commit: {STABLE_GIT_COMMIT}. Built at: 2020-01-22T10:03:59-05:00

1 validators penalized (row 4), 11 rewarded.

![2 penalized 10 rewarded](https://user-images.githubusercontent.com/59260174/72912981-762f4c80-3d0a-11ea-865c-8555f5423cbe.png)

**Comment by prestonvanloon** [comment]
Related #4630

**Comment by rauljordan** [comment]
We consider this resolved, looking at the validator leaderboard, almost everyone is a net positive across the board: https://beaconcha.in/validators/leaderboard. We can close this unless it is still a significant problem.
